lib/tasks/create_coupons.rake

```diff
@@ -0,0 +1,7 @@
+desc "it creates coupon code for all users"
+task create_coupons: :environment do
+  User.find_each do |user|
+    coupon = Coupon.create(code: Faker::Code.asin, description: 'Special Off', start_at: DateTime.now, expire_at: DateTime.now + 1.day, user: user)
```

Comment:
what happens if create fails here?
Difference between start_at and end_at will be slightly more than a day
you can use user.coupons.create here
let's discuss

Reply:
Use current methods instead of now methods on time

Reply:
faker does not guarantee unique code. make sure you generate a unique code everytime

Reply:
user.coupons used.
current methods used.
unique code generated using loop